client/rpc-api/src/chain/mod.rs

```diff
@@ -109,4 +109,24 @@ pub trait ChainApi<Number, Hash, Header, SignedBlock> {
 		metadata: Option<Self::Metadata>,
 		id: SubscriptionId,
 	) -> RpcResult<bool>;
+
+	/// Subscribe to finality events.
+	#[pubsub(
```

Comment:
This should not be part of `chain` API set. We should either make it:
1. a finality-agnostic RPC extension. In which case we need some traits to abstract over different finality gadgets.
2. A grandpa-specific RPC, i which case this should most likely be placed somewhere within `sc-grandpa` directory structure, can access grandpa-specific types directly and is included as RPC extension.

Reply:
I agree with this, it won't be a part of `chain` long term. The issue is I don't want to replicate a bunch of work from #5375 when it comes to setting up a Grandpa specific RPC.